spec/registry.md

```diff
@@ -255,3 +255,548 @@ which only expects the `accord` option:
 > .input {$object :noun case=nominative}
 > {{You see {$color :adjective article=indefinite accord=$object} {$object}!}}
 >```
+
+# Default Registry
+
+This section describes the functions which each implementation MUST provide to be conformant with
+this specification.
+
+## String Value Selection and Formatting
+
+### Functions
+
+The following functions are provided:
+
+The function `:string` provides string selection and formatting.
+
+### Operands
+
+The _operand_ is any literal or an implementation-defined set of string or
+character-sequence types.
+
+> [!NOTE]
+> This should probably include individual character types, such as `char`.
+
+In addition, implementations MAY perform formatting and selection on 
+`operand` values that do not otherwise have a formatting function registered.
+
+### Options
+
+The function `:string` has no options.
+
+> [!NOTE]
+> Proposals for string transformation options or implementation
+> experience with user requirements is desired during the Tech Preview.
+
+### Selection
+
+When implementing [`MatchSelectorKeys`](spec/formatting.md#resolve-preferences), 
+the `:string` selector performs as described below.
+
+- Let `return_value` be a new empty list of strings.
+- Let `operand` be the resolved value of the _operand_.
+  If the `operand` is not a string literal, convert the value to a string literal,
+  or, optionally: emit a _Selection Error_ and return `return_value`.
+- Let `keys` be a list of strings containing keys to match.
+  (Hint: this list is an argument to `MatchSelectorKeys`)
+- For each string `key` in `keys`:
+   - If the value of `key` is equal to the string value of `operand`
+     then `key` matches the selector.
+     A `key` and an `operand` are equal if they consist of the same
+     sequence of Unicode code points.
+     Add `key` to the front of the `return_value` list.
```

Comment:
I think this is wrong. The `return_value` list should have length 0 or 1, per this part of the formatting spec:

"The returned list MUST contain only unique elements of the input list `keys`."

The only way that `return_value` could have length > 1 is if it includes duplicates, since all of the elements in the list must be equal to each other (given the previous step).

Reply:
The list can have duplicates. Consider:

```
.match {$foo :string}{$num :integer}
foo 0   {{You have no foo}}
foo one {{You have {$num} foo}}
foo *   {{You have {$num} foo}}
*   0   {{You have no {$foo}}}
*   one {{You have {$num} {$foo}}}
*   *   {{You have {$num} {$foo}}}
```